mlir/python/requirements.txt

```diff
@@ -1,4 +1,4 @@
-numpy>=1.19.5, <=1.26
-pybind11>=2.9.0, <=2.10.3
+numpy>=1.19.5, <=2.1.2
```

Comment:
How much can we trust the CI checks for this version upgrade? Moving from numpy 1 --> 2 caused (is still causing) quite a bit of ecosystem churn (https://numpy.org/doc/stable/numpy_2_0_migration_guide.html).

Is this a "merge and see if anyone complains" scenario? Or should some other tests/experiments be run or the change be announced on Discourse?

Reply:
Honestly, IDK. I tested this locally and made sure `check-mlir-python` passes but nothing more...

Reply:
I don't think it's our responsibility to keep a lid on this? Anyone that has tighter constraints on numpy will have it in their own `requirements.txt`. The natural way such things get resolved is `pip` does dependency resolution but since we're not distributing a package, it falls to the user to be aware.